### PR TITLE
Iox #509 build iceoryx static lib as default

### LINKED ISSUE #509: iceoryx libs should be build as static libs
## Required information

**Operating system:**
Ubuntu 18.04 LTS

**Compiler version:**
GCC 7.5.0

**Observed result or behaviour:**
When building the iceroyx lib and install them into `build/install/prefix` the application cannot find the .so files because `LD_LIBRARY_PATH` does not point to the custom location.
iox-roudi then throws the error:
```bash
bin/iox-roudi: error while loading shared libraries: libiceoryx_posh_config.so: cannot open shared object file: No such file or directory
```
output from ldd:
```bash
ldd bin/iox-roudi 
        linux-vdso.so.1 (0x00007fff1cf1d000)
        libiceoryx_posh_config.so => not found
        libiceoryx_posh_roudi.so => not found
        libiceoryx_posh.so => not found
        libiceoryx_utils.so => not found
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fb77124c000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fb77122f000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fb77103d000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fb770eee000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fb771474000)
```

**Expected result or behaviour:**
For better usability the default build should be a static lib to avoid setting the `LD_LIBRARY_PATH` manually.
The steps for using iceoryx as shared lib should be documented.

**Conditions where it occurred / Performed steps:**
Build iceoryx with build script and execute:
```bash
./build/install/prefix/bin/iox-roudi
```

### PR DESCRIPTION
**Pull request by dkroenke**
## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Notes for Reviewer
iceoryx is build as static lib per default
in Github CI we build as shared lib to ensure functionality
documentation adapted

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [ ] All checkboxes in the PR checklist are checked or crossed out
1. [ ] Merge

## References

- Closes #509

### CODE REVIEW

File: tools/iceoryx_build_test.sh [hunk 1]
```diff
@@ -183,7 +183,7 @@ while (( "$#" )); do
         echo "    build-all             Build all extensions and all examples"
         echo "    out-of-tree           Out-of-tree build for CI build"
         echo "    build-strict          Build is performed with '-Werror'"
-        echo "    build-static          Build static libs (iceoryx is build as shared lib per default)"
+        echo "    bbuild-shared         Build shared libs (iceoryx is build as static lib per default)"
```
Comment (elBoberido) [inline tools/iceoryx_build_test.sh:186]: typo `bbuild-shared`

**Review by budrus — APPROVED** [review]

**Review by elBoberido — APPROVED** [review]

File: tools/iceoryx_build_test.sh [hunk 2]
```diff
@@ -183,7 +183,7 @@ while (( "$#" )); do
         echo "    build-all             Build all extensions and all examples"
         echo "    out-of-tree           Out-of-tree build for CI build"
         echo "    build-strict          Build is performed with '-Werror'"
-        echo "    build-static          Build static libs (iceoryx is build as shared lib per default)"
+        echo "    build-shared         Build shared libs (iceoryx is build as static lib per default)"
```
Comment (mossmaurice) [inline tools/iceoryx_build_test.sh:186]: Nitpick: Indentation not equal across all lines.

**Review by mossmaurice — APPROVED** [review]

**Comment by elBoberido** [comment]
@dkroenke your last commit is not signed. On the other hand, committer are not required to sign the commits

**Comment by dkroenke** [comment]
> @dkroenke your last commit is not signed. On the other hand, committer are not required to sign the commits
@elBoberido 
Do you mean d344910? That is signed

**Pull request merged**